pyuvdata/uvflag/uvflag.py

```diff
@@ -1286,6 +1589,66 @@ def to_baseline(
                 '"baseline" to match.'
             )
 
+        # write it out this rather than comparing the UVParameters because
+        # future_array_shapes might be different. In the future, when shapes are not
+        # variable, this can be done by comparing the UVParameters.
+        if self.Nfreqs != uv.Nfreqs or not np.allclose(
+            np.squeeze(self.freq_array),
+            np.squeeze(uv.freq_array),
+            rtol=self._freq_array.tols[0],
+            atol=self._freq_array.tols[1],
+        ):
+            raise ValueError(
+                "The freq_array on uv is not the same as the freq_array on this "
+                f"object. The value on this object is {self.freq_array}; the value "
+                f"on uv is {uv.freq_array}"
+            )
+
+        # TODO: should Nants_telescope and Nspws be in this list?
```

Comment:
Nants_telescope I could understand no, but Nspws sounds important. I could imagine setting only the matching spectral windows if they aren't equal. Either way would require more logic right?

Reply:
I think Nspws is getting implicitly checked already because `spw_array` is listed. Nspws is the size of that array, so if the arrays match so do the Nspws. Same is true for `Nants_telescope` since antenna_names/numbers/positions are checked. So I think I'll get rid of this comment.